### PR TITLE
Refactoring render view

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

We have an issue with adding ThirdRGBSensor at evaluation. We do not want to task to have this heavy sensor, so we want to be able to only add it at evaluation. `video_render_views` is only a string, so I replace it with `extra_sim_sensors` that contains the config for the sensors and it will only be used at evaluation.

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/config/default_structured_configs.py [hunk 1]
```diff
@@ -47,6 +49,9 @@ class EvalConfig(HabitatBaselinesBaseConfig):
         # available options are "disk" and "tensorboard"
         default_factory=list
     )
+    extra_sim_sensors: Dict[str, SimulatorSensorConfig] = field(
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/config/default_structured_configs.py:52]: This is meant to replace `video_render_views`. 
`extra_sim_sensors` contains habitat-lab `SimulatorSensorConfig` definitions that will be added at runtime **only when evaluating**

File: habitat-baselines/habitat_baselines/config/rearrange/rl_rearrange.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@
 defaults:
   - /benchmark/rearrange: rearrange
   - /habitat_baselines: habitat_baselines_rl_config_base
+  - /habitat/simulator/sim_sensors@habitat_baselines.eval.extra_sim_sensors.third_rgb_sensor: third_rgb_sensor
```
Comment (vincentpierre) [inline habitat-baselines/habitat_baselines/config/rearrange/rl_rearrange.yaml:6]: This grabs the sim sensor `/habitat/simulator/sim_sensors: third_rgb_sensor` from the config store and places it in `habitat_baselines.eval.extra_sim_sensors.third_rgb_sensor.third_rgb_sensor`. This is a one line change that will replace the `video_render_views` list. 
Adding new evaluation only sensors is as easy as adding sim sensors to the config.

File: habitat-baselines/habitat_baselines/config/default_structured_configs.py (hunk 1 above)
Comment (Skylion007) [inline habitat-baselines/habitat_baselines/config/default_structured_configs.py:52]: I don't like the name extra_sim_sensors? How about eval_sim_sensors?

File: habitat-lab/habitat/utils/gym_definitions.py [hunk 3]
```diff
@@ -64,24 +84,7 @@ def _make_habitat_gym_env(
...
-                )
-            else:
-                default_agent.sim_sensors.update(
-                    {
-                        "default_agent_third_rgb_sensor": ThirdRGBSensorConfig(
-                            uuid="default_robot_third_rgb"
-                        )
-                    }
-                )
+        add_sim_sensor_to_config(config, ThirdRGBSensorConfig())
```
Comment (ASzot) [inline habitat-lab/habitat/utils/gym_definitions.py:87]: Shouldn't this get the sensors from `config.habitat_baselines.eval.extra_sim_sensors` instead?

**Review by ASzot — APPROVED** [review]
Other than the comment about sensors that are added in render mode of the Gym environment, LGTM.

File: habitat-lab/habitat/utils/gym_definitions.py (hunk 3 above)
Comment (vincentpierre) [reply habitat-lab/habitat/utils/gym_definitions.py:87]: When creating the gym environment, the habitat baselines config won't be available, so I added a generic `ThirdRGBSensorConfig` instead.

Comment (ASzot) [reply habitat-lab/habitat/utils/gym_definitions.py:87]: Good point, this makes sense.

**Pull request merged**